ethcore/src/engines/tendermint/mod.rs

```diff
@@ -541,12 +542,21 @@ impl Engine for Tendermint {
 
 	/// Apply the block reward on finalisation of the block.
 	fn on_close_block(&self, block: &mut ExecutedBlock) -> Result<(), Error>{
+		let tracing_enabled = block.tracing_enabled();
 		let fields = block.fields_mut();
+		let mut tracer = ExecutiveTracer::default();
 		// Bestow block reward
 		let reward = self.params().block_reward;
 		let res = fields.state.add_balance(fields.header.author(), &reward, CleanupMode::NoEmpty)
 			.map_err(::error::Error::from)
 			.and_then(|_| fields.state.commit());
+
+		if tracing_enabled {
```

Comment:
@grbIzl This block here is exactly the same for authority round and tendermint, and probably for any new engine we would add in the future. It's also not really the concern of the consensus engine to give out block rewards, so we plan to split that kind of thing out as well. Moving this to a function in `engines::common` module would help us keep the code cleaner and assist in the refactor in the future.